### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by jacobwindsor**
Fixes everything in [#345](https://app.zenhub.com/workspaces/dcp-ingest-product-development-5f71ca62a3cb47326bdc1b5c/issues/ebi-ait/dcp-ingest-central/345) except for the horizontal scrolling issue which I have opened up [#350](https://app.zenhub.com/workspaces/dcp-ingest-product-development-5f71ca62a3cb47326bdc1b5c/issues/ebi-ait/dcp-ingest-central/350) for

### CODE REVIEW

File: src/app/projects/projects.service.ts [hunk 1]
```diff
@@ -14,7 +14,8 @@ export class ProjectsService {
     return this.http.get<any>(this.URL).pipe(
       map((response) => {
         if (response) {
-          return response._embedded.projects.map(this.formatProject);
+          return response._embedded.projects.map(this.formatProject)
+            .filter(project => !!project);
```
Comment (ke4) [inline src/app/projects/projects.service.ts:18]: I had a discussion on this with @jacobwindsor and he told me that he is filtering out the null projects with this filter.
I checked the `formatProject` function and I could only see 1 error throwing when a project has not got title.
Is that really the best solution here to filter out the projects without title? Why can't we show those projects on the UI and maybe grey them out. When the user would hover over with the mouse then we could show a tooltip what is wrong with that project.
In the current solution we only log the error on the console. That can be only seen by a dev but not a wrangler.
@ebi-ait/hca-dev What are other devs thoughts about this?

**Review by ke4 — COMMENTED** [review]
I added a comment, please think about it. Thanks

File: src/app/projects/projects.service.ts (hunk 1 above)
Comment (jacobwindsor) [reply src/app/projects/projects.service.ts:18]: I agree this is probably not the best solution. However, I don't think users should be concerned at all if there is an error with one project. It is not just the missing title that will throw an error. I added that error on purpose as we at least need a title. Other errors would include, if a publication author is missing, if something is a string when it is expected to be an array, etc. There could be a number of errors that could happen during formatting that I cannot predict. This method will catch and log them but the end user will not be aware of them. We don't want the catalogue to break if one project is added in ingest that has unexpected formatting. Any changes pushed to ingest prod could cause the catalogue to break in prod. This was not the case before when we were manually publishing with the `data.json`

Comment (ke4) [reply src/app/projects/projects.service.ts:18]: I think your solution is good, but we won't be notified if there is a problem with any of the projects. Currently we can only check the logs, but we don't do it frequently, I guess.

**Review by ke4 — APPROVED** [review]

File: src/app/projects/projects.service.ts (hunk 1 above)
Comment (aaclan-ebi) [reply src/app/projects/projects.service.ts:18]: Good point, Karoly! I don't think it's critical that we don't get flagged if there is an error in the project and hidden. If someone asked about it we should be able to check the error logs. :) so i guess it's okay!

**Review by aaclan-ebi — APPROVED** [review]

File: src/app/projects/projects.service.ts (hunk 1 above)
Comment (ke4) [reply src/app/projects/projects.service.ts:18]: I approved it, I just wanted to tell my thoughts. I hope it is all right. ;-)

Comment (jacobwindsor) [reply src/app/projects/projects.service.ts:18]: @ke4 of course! Thanks a lot. I think this is good motivation to build centralised logging e.g. Sentry.

**Pull request merged**